src/dstack/_internal/server/services/gateways/__init__.py

```diff
@@ -630,6 +634,15 @@ def gateway_model_to_gateway(gateway_model: GatewayModel) -> Gateway:
 
 
 def _validate_gateway_configuration(configuration: GatewayConfiguration):
+    if configuration.backend not in BACKENDS_WITH_GATEWAY_SUPPORT:
+        raise ServerClientError(
+            f"Gateways are not supported for {configuration.backend.value} backend. "
+            f"Supported backends: {[b.value for b in BACKENDS_WITH_GATEWAY_SUPPORT]}."
+        )
+
+    if configuration.name is not None:
+        validate_dstack_resource_name(configuration.name)
```

Comment:
Does additional validation break backward compatibility?

Reply:
Formally yes, but I don't think it's critical. I consider no name validation at all a bug that should be fixed.